mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -71,6 +71,22 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
 
   /** @group getParam */
   def getItemCol: String = $(itemCol)
+
+  /**
+   * Param for strategy for dealing with unknown users or items at prediction time.
+   * Supported values:
+   * - "nan": prediction value for unknown ids will be NaN.
+   * - "drop": rows in the input DataFrame containing unknown ids will be dropped.
+   * Default: "nan".
+   * @group expertParam
+   */
+  val unknownStrategy = new Param[String](this, "unknownStrategy",
```

Comment:
I don't like the name "unknownStrategy" - seems very ambiguous. I'm not sure what is best, but for example I think "unknownPredictionStrategy" or "newUserItemPredictionStrategy" are better. Just something that will be obvious what it is by the name.